src/Refinery/Factory.php

```diff
@@ -79,4 +77,64 @@ public function string(): String\Group
 	{
 		return new String\Group($this->dataFactory, $this->language);
 	}
+
+	/**
+	 * Contains constraints and transformations for custom functions.
+	 *
+	 * @return Custom\Group
+	 */
+	public function custom(): Custom\Group
+	{
+		return new Custom\Group($this->dataFactory, $this->language);
+	}
+
+	/**
+	 * Contains constraints for container types (e.g. arrays)
+	 *
+	 * @return Container\Group
+	 */
+	public function container()
+	{
+		return new Container\Group($this->dataFactory);
+	}
+
+	/**
+	 * Contains constraints for password strings
+	 *
+	 * @return Password\Group
+	 */
+	public function password()
+	{
+		return new Password\Group($this->dataFactory, $this->language);
+	}
+
+	/**
+	 * Contains constraints for logical compositions with other constraints
+	 *
+	 * @return Logical\Group
+	 */
+	public function logical()
+	{
+		return new Logical\Group($this->dataFactory, $this->language);
+	}
+
+	/**
+	 * Contains constraints for null types
+	 *
+	 * @return Null\Group
+	 */
+	public function null()
+	{
+		return new Null\Group($this->dataFactory, $this->language);
```

Comment:
What about a `state` group instead @legionth / @klees ?

Reply:
I agree with you `null` is not a perfect name for this, but I'm not sure what `state` actual is. Often the term `state` is also used for boolean states, which can lead to confusion IMO. I have no actual  counter proposal. So further input is much appreciated :+1:

Reply:
You are right, this could lead to confusion. So IMO `null` is okay for now, maybe we find a better wording in future.